scripts/py_matter_idl/matter_idl/generators/java/ChipClusters_java.jinja

```diff
@@ -56,68 +94,83 @@
+import javax.annotation.Nullable;
+
+import static chip.devicecontroller.ChipTLVType.*;
+
 public class ChipClusters {
 
-  public interface DefaultClusterCallback {
+  public interface BaseClusterCallback {
+    void onError(Exception error);
+  }
+
+  public interface DefaultClusterCallback extends BaseClusterCallback {
     void onSuccess();
+  }
+
+  public interface BaseAttributeCallback {
     void onError(Exception error);
+    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
-  public interface CharStringAttributeCallback {
+  public interface CharStringAttributeCallback extends BaseAttributeCallback {
     /** Indicates a successful read for a CHAR_STRING attribute. */
     void onSuccess(String value);
-    void onError(Exception error);
-    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
-  public interface OctetStringAttributeCallback {
+  public interface OctetStringAttributeCallback extends BaseAttributeCallback {
     /** Indicates a successful read for an OCTET_STRING attribute. */
     void onSuccess(byte[] value);
-    void onError(Exception error);
-    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
-  public interface IntegerAttributeCallback {
+  public interface IntegerAttributeCallback extends BaseAttributeCallback {
     void onSuccess(int value);
-    void onError(Exception error);
-    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
-  public interface LongAttributeCallback {
+  public interface LongAttributeCallback extends BaseAttributeCallback {
     void onSuccess(long value);
-    void onError(Exception error);
-    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
-  public interface BooleanAttributeCallback {
+  public interface BooleanAttributeCallback extends BaseAttributeCallback {
     void onSuccess(boolean value);
-    void onError(Exception error);
-    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
-  public interface FloatAttributeCallback {
+  public interface FloatAttributeCallback extends BaseAttributeCallback {
     void onSuccess(float value);
-    void onError(Exception error);
-    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
-  public interface DoubleAttributeCallback {
+  public interface DoubleAttributeCallback extends BaseAttributeCallback {
     void onSuccess(double value);
-    void onError(Exception error);
-    default void onSubscriptionEstablished(long subscriptionId) {}
   }
 
   public static abstract class BaseChipCluster {
     protected long chipClusterPtr;
 
-    public BaseChipCluster(long devicePtr, int endpointId) {
-      chipClusterPtr = initWithDevice(devicePtr, endpointId);
+    protected long devicePtr;
+    protected int endpointId;
+    protected long clusterId;
+
+    private Optional<Long> timeoutMillis = Optional.empty();
+
+    public BaseChipCluster(long devicePtr, int endpointId, long clusterId) {
```

Comment:
Why we need to pass the clusterID in constructor? does the clusterID fixed for each cluster?